File: crates/superkick-core/src/review.rs
use serde::{Deserialize, Serialize};

use crate::id::AgentSessionId;

/// Result of a single review agent within a review swarm.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewFinding {
    pub agent_name: String,
    pub session_id: AgentSessionId,
    pub passed: bool,
    pub exit_code: Option<i32>,
}

/// Aggregated result of a review swarm step.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewSwarmResult {
    pub findings: Vec<ReviewFinding>,
    pub total_agents: usize,
    pub passed_count: usize,
    pub failed_count: usize,
    /// Whether the swarm passed the configured findings threshold gate.
    pub gate_passed: bool,
}

File: ui/src/types/review.ts
export interface ReviewFinding {
	agent_name: string
	session_id: string
	passed: boolean
	exit_code: number | null
}

export interface ReviewSwarmResult {
	findings: ReviewFinding[]
	total_agents: number
	passed_count: number
	failed_count: number
	gate_passed: boolean
}
